pkg/tecdsa/retry/retry.go

```diff
@@ -54,7 +54,10 @@ func EvaluateRetryParticipantsForSigning(
 		)
 	}
 	operatorToSeatCount := calculateSeatCount(groupMembers)
-	rand.Seed(seed + int64(retryCount))
```

Comment:
I am reading the commit description explaining this change:

> Multiple concurrent members seeding the global RNG and using it to shuffle
operators list simultaneously cause non-deterministic results. Each call
to `EvaluateRetryParticipantsForSigning` must use a local fresh RNG instance
to work properly.

Was the source of the problem that the `Seed` was called multiple times and every new call, even though it's with the same value, leads to other results of `rand.Shuffle`?

Reply:
The problematic scenario was:
- member 1 seeds the global RNG with seed A1 for attempt 1
- member 2 seeds the global RNG with seed A1 for attempt 1 and does the shuffle before member 1. It gets the list L of excluded members for attempt 1
- list L excludes member 2 for attempt 1 so member 2 prepares for attempt 2 by seeding RNG with seed A2
- member 1 does shuffling for attempt 1 but the RNG was already seeded with A2

Of course, this is the simplified version of the problem and the real course was much more complicated for >30 members doing that logic simultaneously

Reply:
To rephrase and confirm my understanding is correct: previously, `EvaluateRetryParticipantsForSigning` used a global-state `rand`, so it was effectively stateful. At the same time, `EvaluateRetryParticipantsForSigning` could be used by multiple goroutines concurrently. Now, every call to `EvaluateRetryParticipantsForSigning` is stateless, because we initiate a new instance of `rng` with `rand.New`

Is this understanding correct?

I don't think it's blocking this PR but I think it would be great to add a unit test to `retry_test.go` with multiple goroutines that would cover this scenario.

Reply:
Yep, that's correct!